src/sisl/_core/atom.py

```diff
@@ -758,9 +758,40 @@ def Z(self):
         uZ = _a.arrayi([a.Z for a in self.atom])
         return uZ[self.species]
 
-    def index(self, atom):
-        """Return the indices of the atom object"""
-        return (self._species == self.species_index(atom)).nonzero()[0]
+    def index(self, *atoms):
+        """Return indices of atoms of the given atom type.
+
+        Parameters
+        ----------
+        atoms : str, Atom, or lists hereof
```

Comment:
Since we should still use `species_index` (for improved usage), it could also be an int.

Maybe it needs typing, but that code hasn't got much attention, so we'll leave that.

Reply:
So to use that, you'll have to wrap it in a try-catch... :(